### DISCUSSION TITLE
Delete Movie Entry?

### DISCUSSION THREAD

**Discussion by jondough** [IgnisDa/ryot#495] [Q&A] [answered]
Stumbled upon this (https://github.com/IgnisDa/ryot/discussions/321) discussion while trying to delete a test entry I made.

Is there a way we can run the cronjob manually that removes the entries based on the conditions you posted above? I have now waited two days and the test entry I made (through Plex) is still available? Also tried logging into the docker image but there is no sh or bash. Given that the buttons have been lessened to just 4 now on the actions screen would it not be possible to have a red delete button? It seems only right if there is an add button to also have a delete button from a ui perspective.

P.S. - Under Summary the movie count never changes (currently shows 1 Movies, 2 hours 22 minutes) but when I click into the movies it shows four movies found. I've tried to run regenerate summaries but it does not change. Again, these 4 movies have now been there for two days or more.

Running version 3.4.8 with a Postgres db both in docker.

Thanks.

**Comment by IgnisDa** [comment]
I think there's a bigger problem to be addressed than the movies not disappearing from the list pages. Summary calculation not working is a bug.

Can you turn on trace logging `RUST_LOG=ryot=trace` env variable and show me the logs around midnight (or whatever you have set your timezone to)?

ALso the logs when you regenerate the summary.

**Reply by jondough** [reply]
Added the environment variable and ran regenerate summaries. Only a couple of lines were added so I am just attaching them inline.

     2023-11-15T10:03:32.665327Z  INFO ryot: Running version: 3.4.8
     2023-11-15T10:03:32.726609Z  INFO ryot: Using timezone: America/Chicago
     2023-11-15T10:03:32.743107Z  INFO ryot: Listening on: [::]:8000
     2023-11-15T10:04:16.008580Z TRACE ryot::background: Started job: "RecalculateUserSummary"
     2023-11-15T10:04:16.029031Z TRACE ryot::background: Job: "RecalculateUserSummary", Time Taken: 20ms, Successful = true

The movie entry is still there. It's 3am here currently so I will have to wait until tonight before the cron job should run.

**Reply by IgnisDa** [reply]
> It's 3am here currently so I will have to wait until tonight before

You can also change the `TZ` env variable to someplace whose time is around 11:50pm to simulate an early midnight.

**Reply by jondough** [reply]
Have set the time zone to 'Pacific/Niue' so the cron job should run in the next 40 minutes. Looking at the log file the times are always in GMT, is that correct?

     2023-11-15T10:03:32.665327Z  INFO ryot: Running version: 3.4.8
     2023-11-15T10:03:32.726609Z  INFO ryot: Using timezone: America/Chicago
     2023-11-15T10:03:32.743107Z  INFO ryot: Listening on: [::]:8000
     2023-11-15T10:04:16.008580Z TRACE ryot::background: Started job: "RecalculateUserSummary"
     2023-11-15T10:04:16.029031Z TRACE ryot::background: Job: "RecalculateUserSummary", Time Taken: 20ms, Successful = true
     2023-11-15T10:16:31.428022Z  INFO ryot: Running version: 3.4.8
     2023-11-15T10:16:31.469740Z  INFO ryot: Using timezone: Pacific/Niue
     2023-11-15T10:16:31.476049Z  INFO ryot: Listening on: [::]:8000

**Reply by IgnisDa** [reply]
Yes logs are always in GMT.

**Reply by IgnisDa** [reply]
I also suggest you upgrade to the latest version, just in case.

**Reply by jondough** [reply]
Upgraded, cron job should be running soon.

     2023-11-15T10:44:12.444005Z  INFO ryot: Running version: 3.4.9
     2023-11-15T10:44:12.472401Z  INFO ryot: Using timezone: Pacific/Niue
     2023-11-15T10:44:12.476763Z  INFO ryot: Listening on: [::]:8000

**Reply by jondough** [reply]
All 4 movies are still showing up. I have one that is not attached to any user, no reminders or reviews and not monitored. Log attached.

[ryot.log](https://github.com/IgnisDa/ryot/files/13363858/ryot.log)

**Reply by IgnisDa** [reply]
Can you perform a database dump and send it to me? You can email it (ID is in my GH profile).

**Reply by jondough** [reply]
Sure, let me figure out how to connect to the docker instance pg db and I'll do a dump for you.

**Reply by jondough** [reply]
DB dump sent.

**Reply by IgnisDa** [reply]
Not sure what you used to generate the export. Can you use `pg_dump` which generates a single SQL file?

**Reply by jondough** [reply]
I used pg_dump to create a tar file of the db (ex: pg_dump -U postgres -W -F t postgres > db.tar). You should be able to import from that. I'll send you a single sql file just in case you have any issues though.

**Reply by IgnisDa** [reply]
ah yes, you have exported it in tar format, I would prefer the default.

**Comment by IgnisDa** [comment, marked as answer]
@jondough Thanks for bringing this up. There was a bug. I will publish `v3.4.10` soon which should fix the cleanup logic.

Regarding your original question, I don't want to introduce new buttons to the UI. But you can change the schedule in which the cleanup cron job gets triggered by setting the `SCHEDULER_USER_CLEANUP_EVERY` env variable to `1`. Default is `12`, which means it gets triggered every 12 hrs.

**Reply by IgnisDa** [reply]
As for the summary being "incorrect", Ryot only counts media items with 100% progress, so your movies in progress were not being counted. I plan to fix this behavior, but there's some other major refactors I want to address before I get around to it.

**Reply by jondough** [reply]
Just tested with v3.4.10 and it did clean up the movie at 12am. Thanks again for the quick fix.
